app/src/main/java/net/squanchy/schedule/view/ScheduleDayPageView.kt

```diff
@@ -24,18 +24,15 @@ class ScheduleDayPageView @JvmOverloads constructor(
         val layoutManager = SnappingLinearLayoutManager(context)
         setLayoutManager(layoutManager)
         adapter = EventsAdapter(context)
-        setAdapter(adapter)
 
         val horizontalSpacing = resources.getDimensionPixelSize(R.dimen.card_horizontal_margin)
         val verticalSpacing = resources.getDimensionPixelSize(R.dimen.card_vertical_margin)
         addItemDecoration(CardSpacingItemDecorator(horizontalSpacing, verticalSpacing))
     }
 
     fun updateWith(newData: List<Event>, listener: (Event) -> Unit) {
-        val callback = EventsDiffCallback(adapter.events, newData)
-        val diffResult = DiffUtil.calculateDiff(callback, true) // TODO move off the UI thread
-        adapter.updateWith(newData, listener)
-        diffResult.dispatchUpdatesTo(adapter)
+        setAdapterIfNone(adapter.apply { eventClickListener = listener })
```

Comment:
I'd extract the adapter, so we can clearly separate the 2 things (updating the adapter, updating the recyclerview with the new adapter), since the first one has to happen all the time, the second doesn't.

Reply:
True! d9f97c0